addons/account_audit_trail/models/ir_ui_menu.py

```diff
@@ -0,0 +1,11 @@
+from odoo import models
+
+
+class IrUiMenu(models.Model):
+    _inherit = 'ir.ui.menu'
+
+    def _load_menus_blacklist(self):
+        res = super()._load_menus_blacklist()
+        if not any(company.check_account_audit_trail for company in self.env.companies):
```

Comment:
`load_menus` is ormcache'd and doesn't depend on the companies currently selected
The key is `('self._uid', 'debug', keys=('lang',))`, so the best you can do is
```suggestion
        if not any(company.check_account_audit_trail for company in self.env.user.company_ids):
```

Reply:
Right. I see. I changed it to your suggestion then. Not ideal, but should be fine in the majority of cases, I guess.